### DISCUSSION TITLE
Dispara evento no campo no PoDynamicFormField

### DISCUSSION THREAD

**Discussion by Smmartinez** [po-ui/po-angular#974] [Tire suas dúvidas] [answered]
Bom dia galera.... espero que esteja todos bem...
Existe alguma maneira de disparar uma função num campo utilizando o PoDynamicFormField?
Por exemplo, tenho um campo de cep e gostaria de usar a API do viacep para preencher os campos, isso seria possível?

**Comment by jhonyeduardo** [comment]
Boa tarde @Smmartinez 
Você pode utilizar o evento (p-validate), veja nossa documentação: https://po-ui.io/documentation/po-dynamic-form

**Comment by Smmartinez** [comment]
Boa Tarde @jhonyeduardo eu até tentei usar o p-validade, mas esta retornando o erro ERROR TypeError: can't convert undefined to object fiz um teste apenas fixando o valor, desta forma onChangeFields(changeValue): PoDynamicFormValidation {

      if (changeValue.property === 'cependereco') {
        console.log('entrei')
        return {
          value: { endereco: 'teste' },
          fields: [
            { property: 'endereco' }
          ],
          focus: 'endereco'
        };
      }
    }

não seria isso?

**Reply by gabrielbonfim1** [reply]
Cara, você conseguiu resolver isso? eu to recebendo o mesmo erro, fazendo um teste muito parecido

**Reply by gabrielbonfim1** [reply]
Gente para quem tiver o mesmo problema: Consegui resolver, faltava a propriedade [p-value] no html, associei ela com um objeto vazio e pronto.

No HTML : [p-value]="formValue"
No TS : formValue = {}

**Reply by HorselessName** [reply]
Caramba, eu acabei de postar aqui uma pergunta sobre isso, e a sua resposta resolveu o problema...
Que doideira. Vc sabe pq isso ai acontece? Lá na documentação não diz nada sobre o `p-value` ser obrigatório para as propriedades `p-validate` e `p-validate-fields` funcionarem.

https://github.com/po-ui/po-angular/discussions/1875

**Comment by Smmartinez** [comment]
Boa tarde! Alguém teria uma sugestão para esse problema?

**Comment by Smmartinez** [comment, marked as answer]
Bom dia! Esse processo não deveria funcionar?  https://stackblitz.com/edit/po-ui-8beg1p?file=src/app/app.component.ts

**Reply by jhonyeduardo** [reply]
Boa tarde @Smmartinez

Analisando o fonte, como você faz a busca por HTTP e é assincrono, nosso metódo não vai aguardar e para isso precisaremos permitir que o [p-validate] fosse um observable.

Sugiro abrir uma issue de melhoria, solicitando que o validate permita receber Observable, Function e String

**Reply by wsteixeira** [reply]
```
<po-loading-overlay
    [hidden]="!showLoading">
</po-loading-overlay>

onChangeField(changeValue): PoDynamicFormFieldValidation {
  //  console.log('changedValue', changeValue);
  if (changeValue.property === 'zipCode') {
    this.showLoading = true;
    this.zipcodeService.getResource(changeValue.value).subscribe(response => {
      if (response.ok) {
        this.dynamicForm.model['address'] = response.address;
        this.dynamicForm.model['city'] = response.city;
        this.dynamicForm.model['district'] = response.district;
        this.dynamicForm.model['state'] = response.state;
      } else {
        this.poNotification.warning(response.message);
      }
      this.showLoading = false;
    });
  }
  return {};
}

```

**Reply by wsteixeira** [reply]
API https://ws.apicep.com/cep

**Reply by Smmartinez** [reply]
@wsteixeira Bom dia! Estou voltando neste assunto... essa api é sua?

**Reply by wsteixeira** [reply]
Boa tarde,

Não é minha, mas como é publica eu uso nos meus projetos.
